ouroboros-consensus-cardano/test/Test/ThreadNet/TxGen/Cardano.hs

```diff
@@ -1,15 +1,308 @@
+  ) where
+
+import           Control.Exception (assert)
+
+import           Data.Map (Map)
+import qualified Data.Map as Map
+import           Data.Maybe (maybeToList)
+import qualified Data.Sequence.Strict as StrictSeq
+import qualified Data.Set as Set
+import           Data.SOP.Strict
+
+import           Ouroboros.Consensus.Block (SlotNo (..))
+import           Ouroboros.Consensus.Config
+import           Ouroboros.Consensus.HardFork.Combinator.Ledger
+                     (tickedHardForkLedgerStatePerEra)
+import           Ouroboros.Consensus.HardFork.Combinator.State.Types
+                     (currentState, getHardForkState)
+import           Ouroboros.Consensus.HardFork.Combinator.Util.Telescope as Tele
+import           Ouroboros.Consensus.Ledger.Basics (LedgerConfig, LedgerState,
+                     applyChainTick)
+import           Ouroboros.Consensus.NodeId (CoreNodeId (..))
+
+import           Cardano.Binary (toCBOR)
+
+import           Cardano.Crypto (toVerification)
+import           Cardano.Crypto.DSIGN (Ed25519DSIGN)
+import           Cardano.Crypto.DSIGN.Class (SignKeyDSIGN)
+import           Cardano.Crypto.Hash (Blake2b_224)
+import qualified Cardano.Crypto.Signing as Byron
+import           Cardano.Crypto.VRF.Class (SignKeyVRF)
+
+import qualified Cardano.Chain.Common as Byron
+import           Cardano.Chain.Genesis (GeneratedSecrets (..))
+
+import qualified Shelley.Spec.Ledger.Address as SL
+import qualified Shelley.Spec.Ledger.Address.Bootstrap as SL
+import qualified Shelley.Spec.Ledger.BaseTypes as SL
+import qualified Shelley.Spec.Ledger.Coin as SL
+import qualified Shelley.Spec.Ledger.Credential as SL
+import qualified Shelley.Spec.Ledger.Keys as SL
+import qualified Shelley.Spec.Ledger.LedgerState as SL
+import qualified Shelley.Spec.Ledger.Tx as SL
+import qualified Shelley.Spec.Ledger.TxData as SL
+import qualified Shelley.Spec.Ledger.UTxO as SL
+
+import           Ouroboros.Consensus.Shelley.Ledger (GenTx, ShelleyBlock,
+                     mkShelleyTx)
+import           Ouroboros.Consensus.Shelley.Ledger.Ledger (tickedShelleyState)
+import           Ouroboros.Consensus.Shelley.Protocol.Crypto (ADDRHASH, DSIGN,
+                     TPraosCrypto, VRF)
 
 import           Ouroboros.Consensus.Cardano
+import           Ouroboros.Consensus.Cardano.Block (CardanoEras, GenTx (..))
 
+import qualified Test.ThreadNet.Infra.Shelley as Shelley
 import           Test.ThreadNet.TxGen
 
-import           Test.Consensus.Shelley.MockCrypto (TPraosMockCryptoCompatByron)
+data CardanoTxGenExtra c = CardanoTxGenExtra
+  { ctgeByronGenesisKeys :: GeneratedSecrets
+  , ctgeNetworkMagic     :: Byron.NetworkMagic
+  , ctgeShelleyCoreNodes :: [Shelley.CoreNode c]
+  }
+
+instance ( TPraosCrypto c
+           -- These equalities allow use of Byron bootstrap addresses in the
+           -- Shelley era, since @shelley-spec-ledger@ requires Ed25519 and the
+           -- current Byron-to-Shelley translation requires _224.
```

Comment:
```suggestion
           -- current Byron-to-Shelley translation requires Blake2b_224.
```
Or was the omission intentional?

Reply:
It was, but no need to be coy. Clarified, thanks!